### DISCUSSION TITLE
Disk that reads fine with Catweasel MK4 has issues reading with GW F7 Plus v1.11 / fw 1.3

### DISCUSSION THREAD

**Discussion by qbarnes** [keirf/greaseweazle#314] [Q&A] [answered]
I have a 5.25" floppy disk that is a double-sided MFM 80-track which I believe written by an IBM PC (9 sectors/track, 512 bytes/sector, sectors numbered 1-9).  When I read this disk with my Catweasel MK4, all 1440 sectors read flawlessly with no CRC errors or even retries.

When I read the same disk with `gw read --drive 0 --format ibm.720 file.scp`, I get problems like this:
```
Reading c=0-79:h=0-1 revs=2
Format ibm.720
T0.0: IBM MFM (8/9 sectors) from Raw Flux (66815 flux in 333.79ms)
T0.0: IBM MFM (8/9 sectors) from Raw Flux (166984 flux in 834.23ms) (Retry #1.1)
T0.0: IBM MFM (8/9 sectors) from Raw Flux (267147 flux in 1334.68ms) (Retry #1.2)
T0.0: IBM MFM (8/9 sectors) from Raw Flux (367314 flux in 1835.13ms) (Retry #1.3)
T0.0: Giving up: 1 sectors missing
T0.1: IBM MFM (8/9 sectors) from Raw Flux (66794 flux in 333.80ms)
T0.1: IBM MFM (8/9 sectors) from Raw Flux (166934 flux in 834.24ms) (Retry #1.1)
Unknown mark f9
T0.1: IBM MFM (9/9 sectors) from Raw Flux (267066 flux in 1334.69ms) (Retry #1.2)
T1.0: IBM MFM (8/9 sectors) from Raw Flux (66807 flux in 333.80ms)
T1.0: IBM MFM (8/9 sectors) from Raw Flux (166958 flux in 834.25ms) (Retry #1.1)
T1.0: IBM MFM (8/9 sectors) from Raw Flux (267119 flux in 1334.70ms) (Retry #1.2)
T1.0: IBM MFM (8/9 sectors) from Raw Flux (367282 flux in 1835.14ms) (Retry #1.3)
T1.0: Giving up: 1 sectors missing
```

Sometimes I'll even get 2 sectors missing.  Between different runs I'll get different missing sectors and different unknown marks. I've run `gw` with real time scheduling and it doesn't help.

What's odd is before about track 52, only a few tracks will fully read 9 of 9 sectors.  But then starting around track 52, things start to change.  Tracks start reading 9/9 and by track 60, all 100% read 9/9.

I've used two different floppy drives from different manufacturers, but with the same results, so I know the drive isn't related to the issue.

This is the only 80-track, DS/DD disk I have (or that I've found so far), so I don't have another one to try and compare.  Of my 40-track DS/DD IBM PC formatted disks read with `--tracks step=2 --drive 0 --format ibm.360`, they read great, no problems at all.

Are there any tunables I need to tweak to get better reads or other things to try (or might it be a stupid user problem)?

I'm running Linux with a 6.2.9 kernel and Fedora 37.

**Comment by keirf** [comment]
Remove the `--format` option to get a raw dump, and upload somewhere for me to take a look.

**Reply by keirf** [reply]
By the way, it will be nothing to do with the link to the greaseweazle hardware. If there were a scheduling/buffering issue, you would get a clear error about that.

**Reply by qbarnes** [reply]
Do you want a `--raw` read or an `.scp` read without the `--format ...`?

**Reply by keirf** [reply]
Just remove --format and it will automatically be a raw dump. Otherwise with --format and no --raw the SCP image contents is regenerated by Greaseweazle based on specified format.

**Comment by qbarnes** [comment]
While waiting I went ahead and made both (.scp and --raw), so I uploaded both to a repo with them here: https://github.com/qbarnes/gw-data-images.

There's nothing on the original media that matters.  After I read it with my CW, the tracks just seem blank formatted, but I was curious why GW is having issues with it when CW doesn't.

Hopefully, it's not something too weird.  I was just hoping it was something like, "Try tinkering with this setting to see if you get it to fully work."

**Reply by keirf** [reply]
It ought to just work, so looking for oddities in the raw dump is the place to start.

**Comment by keirf** [comment, marked as answer]
Take a look at your Kryuoflux raw dump using HxC tools (HxCFloppyEmulator_soft_beta.zip from here: https://hxc2001.com/download/floppy_drive_emulator/).
Use the Track Analyzer window and select "stream view" (this is a raw view of the flux) you will see unrecoverable damage. If you look at the "Dummy disk" view you can see the damage as an area of disk on both sides.
If you try to read this disk on Catweasel now, I doubt it would do as well. The disk has probably degraded further after being spun up a few times.

**Reply by qbarnes** [reply]
You're absolutely right!  The CW confirmed that in just two days, the disk went from reading perfect to having a 100 some-odd bad sectors!

But the disk shows no signs of any physical damage or wear whatsoever.  However, I think I know what may have happened to it.  #$^%@ magnets!

I found an ipad case with a hidden, strong magnetic clasp right next to where the disk had been sitting for awhile while I changed things with the GW setup.  Me pulling the disk on and off that case's surface probably did it.  At least that floppy had no important information on it, and it sacrificed its life for saving others of its kind.  The case has now been moved far away from being anywhere near any magnetic media.

So it did turn out to be a stupid user problem.  It didn't occur to me to reread it with the CW again to verify.  The errors just seemed so specific and numerous.  I thought it might be some GW weird timing glitch with the way that particular disk was formatted.  It didn't occur to me the disk might have wrangled with a magnet somehow moving it between the CW and the GW.

I rewrote the disk from its original CW capture.  After the rewrite, it read perfect with the GW.

Thank you for the help, and I'm sorry for wasting your time.  :(
